Directory.Build.props

```diff
@@ -49,6 +57,49 @@
     <RepositoryUrl>git://github.com/dotnet/runtime</RepositoryUrl>
     <ProjectUrl>https://github.com/dotnet/runtime</ProjectUrl>
     <LicenseUrl>https://github.com/dotnet/runtime/blob/master/LICENSE.TXT</LicenseUrl>
+    <LicenseFile>$(RepoRoot)LICENSE.TXT</LicenseFile>
+
+    <!-- Indicates this is not an officially supported release. Release branches should set this to false. -->
+    <IsPrerelease>true</IsPrerelease>
+
+    <!-- Experimental packages should not be stable -->
+    <SuppressFinalPackageVersion Condition="'$(SuppressFinalPackageVersion)' == '' and $(MSBuildProjectName.Contains('Experimental'))">true</SuppressFinalPackageVersion>
+    <IsShippingAssembly Condition="$(MSBuildProjectName.Contains('Experimental'))">false</IsShippingAssembly>
+
+    <!-- We don't want Private packages to be shipped to NuGet.org -->
+    <IsShippingPackage Condition="$(MSBuildProjectName.Contains('Private')) and '$(MSBuildProjectExtension)' == '.pkgproj'">false</IsShippingPackage>
+  </PropertyGroup>
+
+  <!-- Language configuration -->
+  <PropertyGroup>
+    <!-- default to allowing all language features -->
+    <LangVersion>latest</LangVersion>
+    <LangVersion Condition="'$(Language)' == 'C#'">preview</LangVersion>
+    <Deterministic>true</Deterministic>
+
+    <!-- Suppress preview message as we are usually using preview SDK versions. -->
+    <SuppressNETCoreSdkPreviewMessage>true</SuppressNETCoreSdkPreviewMessage>
+
+    <!-- Resource naming bug: https://github.com/microsoft/msbuild/issues/4740 -->
+    <EmbeddedResourceUseDependentUponConvention>false</EmbeddedResourceUseDependentUponConvention>
+  </PropertyGroup>
+
+  <!-- Analyzers -->
+  <PropertyGroup>
+    <CodeAnalysisRuleset>$(LibrariesProjectRoot)CodeAnalysis.ruleset</CodeAnalysisRuleset>
```

Comment:
Majority of the C# code outside libraries is not polished to follow all these codeanalysis rules. Should this stay specific to libraries?

Reply:
That file is used by CoreLib, which lives outside libraries, as well. Extracting it into common makes most sense.

Reply:
CoreLib is referencing all its source files from under libraries, so it can reference the codeanalysis ruleset from under libraries as well.

Reply:
> CoreLib is referencing all its source files from under libraries

https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj#L137-L412?

Reply:
I meant to say`CoreLib is referencing a lot of of its source files from under libraries.

Reply:
If the plan is to move all managed projects to libraries than I agree with what you are saying but I don't think that's the plan? Also this property doesn't trigger thousands of errors/warnings but just sets the path to the rulset file in case a project references the analyzers. cc @stephentoub for dicussion.

Reply:
We are going to have many managed projects outside libraries.

Reply:
> just sets the path to the rulset file in case a project references the analyzers

Ok, if this is a link to file with reasonable set of defaults, this can be set globally. Should the file move to a more global place too then?

Reply:
Also, what one has to do to enable use of the default ruleset? Should we also have the references to the analyzer packages, etc. here so that one can enable use of the default ruleset via a single property?

Reply:
> Also, what one has to do to enable use of the default ruleset? Should we also have the references to the analyzer packages, etc. here so that one can enable use of the default ruleset via a single property

AFAIK that's @stephentoub's plan.

> Should the file move to a more global place too then?

I proposed that as well but @stephentoub was against it. I think if we move all logic into common we should move that file as well.

Reply:
I wasn't against it. I just didn't want to expand that PR to consolidate all the logic that's currently spread across several places. I'm happy for all analysis support for managed code in the repo to be consolidated to one top-level location in the repo.

Reply:
I was referring to this comment: https://github.com/dotnet/runtime/pull/59#issuecomment-554401215. Anyways as you just agreed with moving everything analyzer related into root, I'm happy to do so :)

Reply:
Right. If we're only analyzing src in libraries (plus a few corelib files), it shouldn't be at the root. But this discussion suggests we're talking about enabling it for lots of stuff outside of libraries. Also, I think it should be consolidated, but not at the root of the repo. Isn't it better in eng?

Reply:
The [analyzers.props](https://github.com/dotnet/runtime/blob/master/eng/analyzers.props) file already lives in eng. I think it makes most sense to move the property that points to the ruleset file there too. We can move the rulset file there as well, I don't have a strong opinion on that.

Reply:
Sounds like the ruleset reference should just be added to that file and the ruleset put next to it.

Reply:
Perfect.